test/parallel/test-worker-max-count-disabled.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { spawn } = require('child_process');
+const { Worker } = require('worker_threads');
+const { cpus } = require('os');
+
+if (process.argv[2] === 'child') {
+  const cpu_count = cpus().length;
+
+  // This may need to be updated in the future if there are
+  // any other warnings expected. For now, no warnings are
+  // emitted with this code.
+  process.on('warning', common.mustNotCall());
+
+  const expr = `
+    setInterval(() => {}, 10);
+    require("worker_threads").parentPort.postMessage({});
+  `;
+
+  function makeWorker(workers) {
+    return new Promise((res) => {
+      const worker = new Worker(expr, { eval: true });
+      worker.on('message', res);
```

Comment:
Fwiw, this event will fire at pretty much the same time as `worker.on('online')`. I wasn’t sure whether to comment on it before, but seeing how complex this gets in the nested case, being able to remove at least some of the `parentPort.postMessage({});` might be nice.